libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -1570,6 +1606,67 @@ void test_MQTT_SerializePublishHeader( void )
     TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
 }
 
+/* ========================================================================== */
+
+void test_MQTT_SerializeAck( void )
+{
+    uint8_t buffer[ 200 ];
+    uint8_t expectedPacket[ MQTT_PUBLISH_ACK_PACKET_SIZE ];
+    size_t bufferSize = sizeof( buffer );
+    MQTTStatus_t status = MQTTSuccess;
+    MQTTFixedBuffer_t fixedBuffer = { .pBuffer = buffer, .size = bufferSize };
+    uint8_t packetType = MQTT_PACKET_TYPE_PUBACK;
+
+    const uint16_t PACKET_ID = 1;
+    expectedPacket[ 0 ] = packetType;
+    expectedPacket[ 1 ] = 2U;
+    expectedPacket[ 2 ] = PACKET_ID >> 8;
+    expectedPacket[ 3 ] = PACKET_ID & UINT8_MAX;
+
+    /* Verify parameters. */
+    status = MQTT_SerializeAck( NULL, packetType, PACKET_ID );
+    TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );
+
+    status = MQTT_SerializeAck( &fixedBuffer, packetType, 0 );
```

Comment:
missed validation?

Reply:
Are you referring to the reason for the expected failure? I can add in a comment that packet ID 0 is not allowed.

Reply:
Did you miss yo do this step here? TEST_ASSERT_EQUAL_INT( MQTTBadParameter, status );

Reply:
Oh, I didn't see that. Yes